HwOpus service implementation

gdkchan opened the pull request:
This service is needed by games like Touhou Kobuto V, maybe others.
Added a dependency on the Concentus library that is a portable opus decoder written in C#.
Not sure if I will add the *MultiStream stuff introduced on 3.0.0+ on this PR yet.

Gamedev1909 commented:
portal knights now works because of this but crashes when trying to get to the menu if you want the error just ask

gdkchan commented:
You can open an issue for that game as the error is probably not related to this PR.

Gamedev1909 commented:
will do thanks for this pr it was a pain i had 10 games that needed it now its down to itimezone 4 and isystemclock

Gamedev1909 commented:
![sogood](https://user-images.githubusercontent.com/18043258/42076438-7eba8220-7b4b-11e8-9784-75008ba2574a.PNG)
![3d](https://user-images.githubusercontent.com/18043258/42076439-7ecb191e-7b4b-11e8-9285-e23156735334.PNG)
Touhou Kobuto V Burst Battle now goes in game and renders 3D

jduncanator commented on Ryujinx.HLE/OsHle/Services/Aud/IHardwareOpusDecoderManager.cs:
```diff
@@ -0,0 +1,72 @@
+
+        public long GetWorkBufferSize(ServiceCtx Context)
+        {
+            //Note: The sample rate is ignored because it is fixed to 48KHz.
```
The Opus Decoder decodes the Opus stream to the specified sample rate (48KHz here). The reason for 48KHz is because our AudioRenderer currently only supports this sample rate, and resamples all other audio to 48KHz.

gdkchan replied:
The real hwopus only supports 48khz, therefore the hle implementation also only supports 48khz.

gdkchan commented:
Rebased, also added the function that is used by newer games. Fixed a bug reading the buffer data size from the header. The value is big endian, however it was being read as a little endian value. This was causing problems on Pokémon Let's Go, because it would write a really large size due to the value being read on the wrong order, this later caused the game to use the wrong size on a memcpy and corrupting the heap.
